packages/amplify_authenticator/example/lib/authenticator_with_custom_layout.dart

```diff
@@ -0,0 +1,173 @@
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+import 'package:flutter/material.dart';
+
+import 'main.dart';
+
+// A custom authenticator widget with a custom layout
+//
+// This authenticator provides custom views for Sign In and Sign Up,
+// but falls back on the standard views for all other states
+//
+// The custom views for sign up & sign in add an app logo and change the
+// navigation from a TabBar to buttons at the bottom of the screen
+class AuthenticatorWithCustomLayout extends StatelessWidget {
+  const AuthenticatorWithCustomLayout({
+    Key? key,
+    // required this.state,
+    // required this.child,
+  }) : super(key: key);
+
+  // final AuthenticatorState state;
+  // final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Authenticator(
+      // If a builder method is provided the authenticator will
+      // use this to build the authenticator based on the current state.
+      //
+      // state is the current state of the authenticator. It contains the
+      // current auth step (sign in, sign out, etc.), the form data, and
+      // methods for updating this state.
+      //
+      // child is widget that would be displayed without the use of a custom
+      // builder. If currentStep is AuthenticatorStep.authenticated, this will
+      // be the material app. Otherwise this will be a prebuilt widget from
+      // amplify_authenticator to handle the current AuthenticatorStep
+      builder: (context, state, child) {
```

Comment:
I think `child` is a confusing name - I'd prefer `defaultChild` or `defaultScreen` or just not even pass it in at all and let the method return `null` since every implementation of this should just return it as the default option and will never wrap it.